routemaster/services/ingest.rb

```diff
@@ -13,27 +13,47 @@ class Ingest
       include Mixins::Assert
       include Mixins::Counters
 
-      def initialize(topic:, event:, queue:)
+      def initialize(topic:, event:, queue:, subscriber_name: nil)
         _assert(event.topic == topic.name)
         @topic = topic
         @event = event
         @queue = queue
+        @subscriber_name = subscriber_name
       end
 
       def call
-        data = Services::Codec.new.dump(@event)
-        @topic.subscribers.each do |s|
-          batch = Models::Batch.ingest(data: data, timestamp: @event.timestamp, subscriber: s)
+        publish_to_subscribers
+        update_counters
+        self
+      end
+
+      private
 
-          job = Models::Job.new(name: 'batch', args: batch.uid, run_at: batch.deadline)
-          @queue.push(job)
-          @queue.promote(job) if batch.full?
-        end
+      def data
+        @_data ||= Services::Codec.new.dump(@event)
+      end
 
+      def update_counters
         _counters.incr('events.published', topic: @topic.name)
         _counters.incr('events.bytes', topic: @topic.name, count: data.length)
         @topic.increment_count
-        self
+      end
+
+      def publish_to_subscribers
+        @topic.subscribers.each { |s| publish_to(s) if should_publish_to?(s) }
+      end
+
+      def should_publish_to?(subscriber)
```

Comment:
This method isn't reused so I'd suggest inlining it into `publish_to_subscribers`. There's a significant cost to each method call in Ruby and we've been trying to keep latency as low as possible without compromising readability — hence the slightly larger methods than is typically usual in Rubyland.

Reply:
I see. it makes sense